backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/CpuPinningPolicyUnit.java

```diff
@@ -74,18 +75,27 @@ public CpuPinningPolicyUnit(PolicyUnit policyUnit,
         switch (vm.getCpuPinningPolicy()) {
             case MANUAL:
             case RESIZE_AND_PIN_NUMA:
-                if (StringUtils.isEmpty(cpuPinning)) {
+                if (StringUtils.isEmpty(cpuPinning) && vm.getCpuPinningPolicy() == CpuPinningPolicy.MANUAL) {
                     // The logic below applies to MigrateVm in which we get here with CPU pinning that was set on the
                     // source host. On RunVm, we get here before determining the CPU pinning for the VM so the dynamic
                     // CPU pinning is not set and hosts should not be filtered by the logic below.
                     return hosts;
                 }
-                // collect all pinned host cpus and merge them into one set
-                final Set<Integer> pinnedCpus = CpuPinningHelper.getAllPinnedPCpus(cpuPinning);
+                Set<Integer> pinnedCpus = new LinkedHashSet<>();
+                if (!StringUtils.isEmpty(cpuPinning)) {
+                    // collect all pinned host cpus and merge them into one set
+                    pinnedCpus = CpuPinningHelper.getAllPinnedPCpus(cpuPinning);
+                }
                 for (final VDS host : hosts) {
                     final Collection<Integer> onlineHostCpus = SlaValidator.getOnlineCpus(host);
                     var cpuTopology = resourceManager.getVdsManager(host.getId()).getCpuTopology();
                     final Collection<Integer> dedicatedCpus = cpuTopology.stream().filter(VdsCpuUnit::isDedicated).map(VdsCpuUnit::getCpu).collect(Collectors.toList());
+                    if (!dedicatedCpus.isEmpty() && vm.getCpuPinningPolicy() == CpuPinningPolicy.RESIZE_AND_PIN_NUMA) {
+                        messages.addMessage(host.getId(), EngineMessage.VAR__DETAIL__VM_PINNING_CANT_RESIZE_WITH_DEDICATED.name());
+                        log.debug("Host {} does not satisfy CPU pinning constraints, cannot match virtual topology " +
+                                "with available CPUs.", host.getId());
```

Comment:
why not to have the message informative here - to say that we can't resize the VM due to exclusively pinned cpus on the host (I see now that you wrote it below - for the audit message)